### DISCUSSION TITLE
what is this fork URL??

### DISCUSSION THREAD

**Discussion by 0xconstructor** [Cyfrin/foundry-full-course-cu#31] [Q&A] [answered]
![Screenshot (39)](https://github.com/Cyfrin/foundry-full-course-f23/assets/123366816/9668e652-7beb-4167-bdbd-ac4a5e76706c)

**Comment by cromewar** [comment]
Hello @web3baji Please describe a little bit more your question so we can provide better support.

**Reply by 0xconstructor** [reply]
why do I get that error message indicates that a value is required for the --fork-url option, but it was not supplied in the command ,even though I have been doing the exactly as video suggests ,so how do I deploy the code when it always pops out the error that fork URL is missing ,please help me to resolve the issue

**Reply by 0xconstructor** [reply]
![Screenshot (41)](https://github.com/Cyfrin/foundry-full-course-f23/assets/123366816/c634fecc-8b14-48cb-9fa5-3f1215bdc6cb)

whether it was deploying to sepolia network or the local network it results in this error that fork URL flag is missing ,so how do I rewrite the code or what changes do I need to make to correct that error

**Reply by AztecWarlord** [reply]
if you did not add the `$SEPOLIA_RPC_URL` to your terminal. Refer to video 2 @30:41 to add it, otherwise when running the script `$SEPOLIA_RPC_URL` called will be empty. Good luck!

**Reply by cromewar** [reply]
Does @AztecWarlord advice worked your you @web3baji ?

**Comment by cromewar** [comment]
The reason is that $RPC_URL is a env variable, don't forget to add it on your console with:

`export RPC_URL=your_IP`

or you can just put it manually instead of using the `env` variable, the choice is yours.

**Comment by alymurtazamemon** [comment]
@web3baji Follow these steps;

1. Define these variables inside `.env` file;
```env
SEPOLIA_RPC_URL=https://eth-sepolia.g.alchemy.com/v2/...
PRIVATE_KEY=...
```
Update your keys here;
2. Go to the .toml file and define these below;
```toml
[rpc_endpoints]
sepolia = "${SEPOLIA_RPC_URL}"
```
3. Go to the terminal and run `source .env`
4. Run this command;
```bash
forge script DeploySimpleStorage --rpc-url sepolia --private-key $PRIVATE_KEY --broadcast
```

Let me know the result

**Reply by Perso77** [reply]
advice worked

**Comment by 0xRickey** [comment, marked as answer]
My problem was that I didn't save my .env file. Rookie mistake.

**Reply by cromewar** [reply]
Don't worry, errors happen.

**Reply by PatrickAlphaC** [reply]
At least you didn't push it to GitHub :)
